plugins/woocommerce/src/Internal/DataStores/Orders/LegacyDataHandler.php

```diff
@@ -321,28 +323,87 @@ public function get_order_from_datastore( int $order_id, string $data_store_id =
 	 * @param int    $order_id               Order ID.
 	 * @param string $source_data_store      Datastore to use as source. Should be either 'hpos' or 'posts'.
 	 * @param string $destination_data_store Datastore to use as destination. Should be either 'hpos' or 'posts'.
+	 * @param array  $fields                 List of metakeys or order properties to limit the backfill to.
 	 * @return void
 	 * @throws \Exception When an error occurs.
 	 */
-	public function backfill_order_to_datastore( int $order_id, string $source_data_store, string $destination_data_store ) {
+	public function backfill_order_to_datastore( int $order_id, string $source_data_store, string $destination_data_store, array $fields = array() ) {
 		$valid_data_stores = array( 'posts', 'hpos' );
 
 		if ( ! in_array( $source_data_store, $valid_data_stores, true ) || ! in_array( $destination_data_store, $valid_data_stores, true ) || $destination_data_store === $source_data_store ) {
 			throw new \Exception( sprintf( 'Invalid datastore arguments: %1$s -> %2$s.', $source_data_store, $destination_data_store ) );
 		}
 
-		$order = $this->get_order_from_datastore( $order_id, $source_data_store );
-
-		switch ( $destination_data_store ) {
-			case 'posts':
-				$order->get_data_store()->backfill_post_record( $order );
-				break;
-			case 'hpos':
-				$this->posts_to_cot_migrator->migrate_orders( array( $order_id ) );
-				break;
-			default:
-				break;
+		$fields    = array_filter( $fields );
+		$src_order = $this->get_order_from_datastore( $order_id, $source_data_store );
+
+		// Backfill entire orders.
+		if ( ! $fields ) {
+			if ( 'posts' === $destination_data_store ) {
+				$src_order->get_data_store()->backfill_post_record( $src_order );
+			} elseif ( 'hpos' === $destination_data_store ) {
+				$this->posts_to_cot_migrator->migrate_orders( array( $src_order->get_id() ) );
+			}
+
+			return;
+		}
+
+		$this->validate_backfill_fields( $fields, $src_order );
+
+		$dest_order = $this->get_order_from_datastore( $src_order->get_id(), $destination_data_store );
+
+		if ( 'posts' === $destination_data_store ) {
+			$datastore = $this->data_store->get_cpt_data_store_instance();
+		} elseif ( 'hpos' === $destination_data_store ) {
+			$datastore = $this->data_store;
+		}
+
+		if ( ! $datastore || ! method_exists( $datastore, 'update_order_from_object' ) ) {
+			throw new \Exception( __( 'The backup datastore does not support updating orders.', 'woocommerce' ) );
+		}
+
+		// Backfill meta.
+		if ( ! empty( $fields['meta_keys'] ) ) {
+			foreach ( $fields['meta_keys'] as $meta_key ) {
+				$dest_order->delete_meta_data( $meta_key );
+
+				foreach ( $src_order->get_meta( $meta_key, false, 'edit' ) as $meta ) {
+					$dest_order->add_meta_data( $meta_key, $meta->value );
+				}
+			}
 		}
+
+		// Backfill props.
+		if ( ! empty( $fields['props'] ) ) {
+			$new_values = array_combine(
+				$fields['props'],
+				array_map(
+					fn( $prop_name ) => $src_order->{"get_{$prop_name}"}(),
```

Comment:
Should we check to see if $src_order does indeed have that method before calling it?

Reply:
Just saw that `validate_backfill_fields` already ensures this.